xarray/core/dataset.py

```diff
@@ -8424,6 +8485,36 @@ def argmin(self: T_Dataset, dim: Hashable | None = None, **kwargs) -> T_Dataset:
         -------
         result : Dataset
 
+        Example
+        -------
+
+        import xarray as xr
+
+        dataset = xr.Dataset(
+            {
+                'math_scores': (
+                    ['student', 'test'],
+                    [[90, 85, 79],
+                    [78, 80, 85],
+                    [95, 92, 98]]
+                ),
+                'english_scores': (
+                    ['student', 'test'],
+                    [[88, 80, 92],
+                    [75, 95, 79],
+                    [93, 96, 78]]
+                )
+            },
+            coords={
+                'student': ['Alice', 'Bob', 'Charlie'],
+                'test': ['Test 1', 'Test 2', 'Test 3']
+            }
+        )
+
+        argmin_indices = dataset.argmin(dim='student')
```

Comment:
Can add text explaining this

Can also feed the resulting indexes back in to `.isel` (that's why this `argmin` method exists)

Might be good to think of other example datasets for which you would be really interested in the index.